microceph/ceph/bootstrap.go

```diff
@@ -102,10 +112,66 @@ func Bootstrap(s common.StateInterface) error {
 		return err
 	}
 
+	// Configure defaults cluster configs for network.
+	err = setDefaultNetworks(data.ClusterNet)
+	if err != nil {
+		return err
+	}
+
 	// Re-generate the configuration from the database.
 	err = UpdateConfig(s)
 	if err != nil {
-		return fmt.Errorf("Failed to re-generate the configuration: %w", err)
+		return fmt.Errorf("failed to re-generate the configuration: %w", err)
+	}
+
+	return nil
+}
+
+// setDefaultNetworks configures the public and cluster networks on mon KV store.
+func setDefaultNetworks(cn string) error {
+	// Cluster Network
+	err := SetConfigItem(apiTypes.Config{
+		Key:   "cluster_network",
+		Value: cn,
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func prepareCephBootstrapData(s common.StateInterface, data *common.BootstrapConfig) error {
+	var err error
+
+	// if no mon-ip is provided.
+	if len(data.MonIp) == 0 {
+		if len(data.PublicNet) == 0 {
+			// Use default value
+			data.MonIp = s.ClusterState().Address().Hostname()
+		} else {
+			data.MonIp, err = common.Network.FindIpOnSubnet(data.PublicNet)
+			if err != nil {
+				return fmt.Errorf("failed to locate %s on host: %w", data.MonIp, err)
+			}
+		}
+	}
+
+	// If public network value is provided by user
+	if len(data.PublicNet) != 0 {
```

Comment:
hmm, by the time the control reaches that line 161, there WILL BE a value for MonIp, either deduced from public network or the default from MicroCluster. This structure  is chosen because any or all of the parameters may or may not be provided by the user, but the bootstrap process expects all of them, i.e. The logic either calculates appropriate values or falls back to defaults.